sway/config.c

```diff
@@ -647,6 +647,24 @@ void run_deferred_commands(void) {
 	transaction_commit_dirty();
 }
 
+void run_deferred_bindings(void) {
+	struct sway_seat *seat;
+	wl_list_for_each(seat, &(server.input->seats), link) {
+		if (!seat->deferred_bindings->length) {
+			continue;
+		}
+		sway_log(SWAY_DEBUG, "Running deferred bindings for seat %s",
+				seat->wlr_seat->name);
+		while (seat->deferred_bindings->length) {
+			struct sway_binding *binding = seat->deferred_bindings->items[0];
+			seat_execute_command(seat, binding);
+			list_del(seat->deferred_bindings, 0);
+			free_sway_binding(binding);
+		}
+	}
+	transaction_commit_dirty();
```

Comment:
Maybe we can move this out of this function, and add it after this function's call, so that we atomically apply the new state for both deferred commands and bindings?

Reply:
Done